File: manual/code/generate_strategies/IR(summary)/python/python/function_itertools_permutations.py
from itertools import permutations
from collections import defaultdict


class PermutationsProcessor:

    def get_permutations(self, data):
        if not isinstance(data, str):
            raise ValueError("Input data must be a string.")

        
        perms = [''.join(p) for p in sorted(permutations(data))]

        
        perm_dict = defaultdict(int)
        for perm in perms:
            perm_dict[perm] += 1

        
        perms_str = ', '.join(perms)
        perm_dict_str = ', '.join(f"{key}: {value}" for key, value in perm_dict.items())

        return perms_str + "\n" + perm_dict_str

File: manual/code/generate_strategies/IR(summary)/java/java/function_itertools_permutations.java
package com.example;

import org.apache.commons.collections4.iterators.PermutationIterator;
import java.util.*;
import java.util.stream.Collectors;

public class PermutationsProcessor {
    public String get_permutations(String data) {
        if (data == null) {
            throw new IllegalArgumentException("Input data must be a string.");
        }

        List<Character> chars = data.chars()
                .mapToObj(ch -> (char) ch)
                .collect(Collectors.toList());

        
        PermutationIterator<Character> permutations = new PermutationIterator<>(chars);
        List<String> permsList = new ArrayList<>();
        Map<String, Integer> freqMap = new HashMap<>();

        
        while (permutations.hasNext()) {
            String perm = permutations.next().stream()
                    .map(String::valueOf)
                    .collect(Collectors.joining());
            permsList.add(perm);
            freqMap.merge(perm, 1, Integer::sum);
        }

        
        Collections.sort(permsList);

        
        String permsStr = String.join(", ", permsList);
        String freqStr = freqMap.entrySet().stream()
                .sorted(Map.Entry.comparingByKey())
                .map(entry -> entry.getKey() + ": " + entry.getValue())
                .collect(Collectors.joining(", "));

        return permsStr + "\n" + freqStr;
    }
}